Remotion/Web/UnitTests/Core/WebStringTest.cs

```diff
@@ -0,0 +1,210 @@
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Attribute));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromRawAttribute_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromRawAttribute (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void CreateFromRawHtml ()
+    {
+      var stringValue = "<a>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Html));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromRawHtml_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromRawHtml (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void CreateFromString ()
+    {
+      var stringValue = "My <b>test</a> string";
+      var webString = WebString.CreateFromString (stringValue);
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromString_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromString (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void ImplicitConversionToString ()
+    {
+      var stringValue = "My <b>test</a> string";
+      WebString webString = stringValue;
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void IsEmpty ()
+    {
+      Assert.That (new WebString().IsEmpty, Is.True);
+      Assert.That (((WebString)"").IsEmpty, Is.True);
+      Assert.That (((WebString)" ").IsEmpty, Is.False);
+      Assert.That (((WebString)"test").IsEmpty, Is.False);
+    }
+
+    [Test]
+    public void Write_Text ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromString (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo ("My &lt;b&gt;test&lt;/a&gt;"));
+    }
+
+    [Test]
+    public void Write_Html ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void Write_Attribute_ThrowsInvalidOperationException ()
+    {
+      var stringValue = "My <b>test</a>";
+      var webString = WebString.CreateFromRawAttribute (stringValue);
+
+      Assert.That (() => ExecuteWithHtmlTextWriter (webString.Write), Throws.InvalidOperationException);
```

Comment:
Please assert ex message too. Ex-messages that we write outselves and are meant to comminucate important informaton / bug-details are usually part of the tested contract.